lib/active_model/serializer/caching.rb

```diff
@@ -256,61 +247,53 @@ def cache_check(adapter_instance)
       #   When +name+ is <tt>User::Admin</tt>
       #   creates the Serializer classes (if they don't exist).
       #     CachedUser_AdminSerializer
-      #     NonCachedUser_AdminSerializer
       #
       # Given a hash of its cached and non-cached serializers
       # 1. Determine cached attributes from serializer class options
       # 2. Add cached attributes to cached Serializer
       # 3. Add non-cached attributes to non-cached Serializer
       def fetch_fragment_cache(adapter_instance)
-        serializer_class_name = self.class.name.gsub('::'.freeze, '_'.freeze)
         self.class._cache_options ||= {}
         self.class._cache_options[:key] = self.class._cache_key if self.class._cache_key
 
-        cached_serializer = _get_or_create_fragment_cached_serializer(serializer_class_name)
+        cached_serializer = _get_or_create_fragment_cached_serializer
         cached_hash = ActiveModelSerializers::SerializableResource.new(
           object,
           serializer: cached_serializer,
           adapter: adapter_instance.class
         ).serializable_hash
```

Comment:
first pass at trying to remove the extra class failed, so I left it here

something like

```ruby
        cached_hash = cached_attributes.fetch(key) do
          self.class.cache_store.fetch(key) do
            attributes(fields, true)
          end
        end
```

(added `cached_attributes` to the method params and removed all references to `_fragmented` since there would be no more special classes)